test/test_endpoints.py

```diff
@@ -350,3 +381,147 @@ def test_get_transaction_trace_of_rejected():
     assert resp.status_code == 500
+
+
+@devnet_in_background()
+def test_send_message_to_l2_deploy_execute():
+    """Test POST l1 to l2 deploy contract and execute transaction"""
+    # Deploy L1L2 contract
+    deploy_info = deploy(contract=L1L2_CONTRACT_PATH)
+
+    # Create l1 to l2 mock transaction
+    response = send_message_to_l2(
+        {
+            "l2_contract_address": deploy_info["address"],
+            "entry_point_selector": "0xC73F681176FC7B3F9693986FD7B14581E8D540519E27400E88B8713932BE01",
+            "l1_contract_address": "0xe7f1725E7734CE288F8367e1Bb143E90bb3F0512",
+            "payload": ["0x1", "0x1"],
+            "nonce": "0x0",
+        }
+    )
+
+    # Check balace of user
+    value = call(
+        function="get_balance",
+        address=deploy_info["address"],
+        abi_path=L1L2_ABI_PATH,
+        inputs=[str(USER_ID)],
+    )
+
+    assert int(value) == 1
+    assert response.status_code == 200
+    assert_tx_status(hex(response.json().get("invoke_tx_hash")), "ACCEPTED_ON_L2")
+
+
+@devnet_in_background()
+def test_send_message_to_l2_execute_without_data():
+    """Test POST l1 to l2 without data"""
+    # Create l1 to l2 mock transaction
+    response = send_message_to_l2(
+        {
+            "l2_contract_address": "",
+            "entry_point_selector": "",
+            "l1_contract_address": "",
+            "payload": "",
+            "nonce": "",
+        }
+    )
+
+    assert response.status_code == 400
+    assert response.json().get("code") == str(StarkErrorCode.MALFORMED_REQUEST)
+
+
+@devnet_in_background()
+def test_send_message_to_l2_execute_without_deploy():
+    """Test POST l1 to l2 without the target contract being deployed"""
+    # Create l1 to l2 mock transaction
+    response = send_message_to_l2(
+        {
+            "l2_contract_address": "0x00285ddb7e5c777b310d806b9b2a0f7c7ba0a41f12b420219209d97a3b7f25b2",
+            "entry_point_selector": "0xC73F681176FC7B3F9693986FD7B14581E8D540519E27400E88B8713932BE01",
+            "l1_contract_address": "0xe7f1725E7734CE288F8367e1Bb143E90bb3F0512",
+            "payload": ["0x1", "0x1"],
+            "nonce": "0x0",
+        }
+    )
+
+    assert response.status_code == 200
+    assert_tx_status(hex(response.json().get("invoke_tx_hash")), "REJECTED")
+
+
+@devnet_in_background(*PREDEPLOY_ACCOUNT_CLI_ARGS)
+def test_send_message_to_l1_deploy_execute():
+    """Test POST l2 to l1 deploy contract and execute transaction"""
+    deploy_info = deploy(L1L2_CONTRACT_PATH)
+
+    # increase and withdraw balance
+    invoke(
+        calls=[(deploy_info["address"], "increase_balance", [USER_ID, 3333])],
+        account_address=PREDEPLOYED_ACCOUNT_ADDRESS,
+        private_key=PREDEPLOYED_ACCOUNT_PRIVATE_KEY,
+    )
+    invoke(
+        calls=[
+            (
+                deploy_info["address"],
+                "withdraw",
+                [USER_ID, 1000, 0xE7F1725E7734CE288F8367E1BB143E90BB3F0512],
```

Comment:
Can you extract the L1 contract address to a variable? Can be a `VARIABLE` in this file since it's only used here.

Reply:
I refactored constants a bit in https://github.com/Shard-Labs/starknet-devnet/pull/365/commits/e1b2f1ecdbdfccee04592077d2fbb241be9cf982